core/src/test/java/org/javarosa/core/model/test/FormDefTest.java

```diff
@@ -432,7 +440,49 @@ public void testSimilarBindConditionsAreDistinguished() throws Exception {
         }
     }
 
-
+    // regression test for when we weren't decrementing multiplicities correctly when repeats were deleted
+    @Test
+    public void testDeleteRepeatMultiplicities() throws IOException {
+        FormParseInit fpi = new FormParseInit("/multiple_repeats.xml");
+        FormEntryController fec = initFormEntry(fpi, "en");
+        fec.stepToNextEvent();
+        fec.newRepeat();
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Repeat 1/1"));
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Repeat 1/2"));
+        fec.stepToNextEvent();
+        fec.newRepeat();
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Repeat 2/1"));
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Repeat 2/2"));
+        fec.stepToNextEvent();
+        fec.stepToNextEvent();
+        fec.newRepeat();
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Repeat 3/1"));
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Repeat 3/2"));
+        fec.stepToNextEvent();
+        fec.newRepeat();
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Repeat 4/1"));
+        fec.stepToNextEvent();
+        fec.answerQuestion(new StringData("Repeat 4/2"));
+        fec.stepToPreviousEvent();
+        fec.stepToPreviousEvent();
+
+        fec.deleteRepeat(0);
+        TreeElement root = fpi.getFormDef().getInstance().getRoot();
+
+        // Confirm that the deleted repeat is gone and its sibling's multiplicity reduced
+        assert root.getChildMultiplicity("question4") == 1;
+        assert root.getChild("question1", 4) == null;
+        // Confirm that the other repeat is unchanged
+        assert root.getChildMultiplicity("question1") == 2;
+        assert root.getChild("question1", 1) != null;
```

Comment:
Can you add similar checks before `deleteRepeat` so we can more easily see what has changed?